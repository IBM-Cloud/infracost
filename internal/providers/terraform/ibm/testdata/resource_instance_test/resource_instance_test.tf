terraform {
  required_providers {
    ibm = {
      source = "IBM-Cloud/ibm"
      version = "1.63.0"
    }
  }
}

provider "ibm" {
    region = "us-south"
}

resource "ibm_resource_instance" "resource_instance_kms" {
  name              = "test"
  service           = "kms"
  plan              = "tiered-pricing"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "resource_instance_secrets_manager" {
  name              = "test"
  service           = "secrets-manager"
  plan              = "standard"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "resource_instance_appid" {
  name              = "test"
  service           = "appid"
  plan              = "graduated-tier"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "resource_instance_power_iaas" {
  name              = "test"
  service           = "power-iaas"
  plan              = "power-virtual-server-group"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "resource_instance_logdna_lite" {
  name              = "logdna-lite"
  service           = "logdna"
  plan              = "lite"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_logdna_7day" {
  name              = "logdna-7day"
  service           = "logdna"
  plan              = "7-day"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_logdna_7day_no_usage" {
  name              = "logdna-7day-no-usage"
  service           = "logdna"
  plan              = "7-day"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_activity_tracker_lite" {
  name              = "activity-tracker-lite"
  service           = "logdnaat"
  plan              = "lite"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_activity_tracker_7day" {
  name              = "activity-tracker-7day"
  service           = "logdnaat"
  plan              = "7-day"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_activity_tracker_7day_no_usage" {
  name              = "activity-tracker-7day-no-usage"
  service           = "logdnaat"
  plan              = "7-day"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_monitoring_lite" {
  name              = "sysdig-lite"
  service           = "sysdig-monitor"
  plan              = "lite"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_monitoring_graduated" {
  name              = "sysdig-graduated"
  service           = "sysdig-monitor"
  plan              = "graduated-tier"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_monitoring_graduated_no_usage" {
  name              = "sysdig-graduated-no-usage"
  service           = "sysdig-monitor"
  plan              = "graduated-tier"
  location          = "us-south"
}

resource "ibm_resource_instance" "resource_instance_monitoring_graduated_secure" {
  name              = "sysdig-graduated-secure"
  service           = "graduated-tier-sysdig-secure-plus-monitor"
  plan              = "7-day"
  location          = "us-south"
}

resource "ibm_resource_instance" "cd_instance_professional" {
  name              = "cd_professional"
  service           = "continuous-delivery"
  plan              = "professional"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "cd_instance_lite" {
  name              = "cd_lite"
  service           = "continuous-delivery"
  plan              = "lite"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wml_instance_lite" {
  name              = "wml_lite"
  service           = "pm-20"
  plan              = "lite"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wml_instance_essentials" {
  name              = "wml_essentials"
  service           = "pm-20"
  plan              = "v2-standard"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wml_instance_standard" {
  name              = "wml_standard"
  service           = "pm-20"
  plan              = "v2-professional"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wa_instance_lite" {
  name              = "wa_lite"
  service           = "conversation"
  plan              = "lite"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wa_instance_trial" {
  name              = "wa_trial"
  service           = "conversation"
  plan              = "plus-trial"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wa_instance_plus" {
  name              = "wa_plus"
  service           = "conversation"
  plan              = "plus"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "wa_instance_enterprise" {
  name              = "wa_enterprise"
  service           = "conversation"
  plan              = "enterprise"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "watson_discovery_plus" {
  name              = "wd_plus"
  service           = "discovery"
  plan              = "plus"
  location          = "us-south"
  resource_group_id = "default"
}

resource "ibm_resource_instance" "watson_discovery_enterprise" {
  name              = "wd_enterprise"
  service           = "discovery"
  plan              = "enterprise"
  location          = "us-south"
  resource_group_id = "default"
}
