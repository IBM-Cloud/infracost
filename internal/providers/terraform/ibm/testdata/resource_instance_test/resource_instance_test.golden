
 Name                                                                              Monthly Qty  Unit                            Monthly Cost 
                                                                                                                                             
 ibm_resource_instance.resource_instance_activity_tracker_7day                                                                               
 └─ Gigabyte Months                                                                         10  Gigabyte Months                       $15.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_activity_tracker_7day_no_usage                                                                      
 └─ Gigabyte Months                                                      Monthly cost depends on usage: $1.50 per Gigabyte Months            
                                                                                                                                             
 ibm_resource_instance.resource_instance_activity_tracker_lite                                                                               
 └─ Gigabyte Months                                                                         10  Gigabyte Months                        $0.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_appconnect_ent                                                                                      
 ├─ Flow runs                                                                               10  1k runs                              $400.00 
 ├─ Gigabytes transmitted outbounds                                                         10  GB                                   $200.00 
 └─ VCPU                                                                                    10  VCPU hours                            $42.50 
                                                                                                                                             
 ibm_resource_instance.resource_instance_appconnect_lite                                                                                     
 └─ Lite plan                                                                                1                                         $0.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_appconnect_pro                                                                                      
 ├─ Flow runs                                                                               10  1k runs                              $400.00 
 └─ Gigabytes transmitted outbounds                                                         10  GB                                   $200.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_appid                                                                                               
 ├─ Users (first 1 Users)                                                                    1  Users                                  $0.00 
 ├─ Users (next 9 Users)                                                                     9  Users                                  $0.02 
 ├─ Users (next 90 Users)                                                                   90  Users                                  $0.14 
 ├─ Users (next 900 Users)                                                                 900  Users                                  $1.26 
 ├─ Users (next 4000 Users)                                                                  0  Users                                  $0.00 
 ├─ Users (over 5000 Users)                                                                  0  Users                                  $0.00 
 ├─ Authentications (first 1 Authentications)                                                1  Authentications                        $0.00 
 ├─ Authentications (next 9 Authentications)                                                 9  Authentications                        $0.03 
 ├─ Authentications (next 90 Authentications)                                               90  Authentications                        $0.28 
 ├─ Authentications (next 900 Authentications)                                             900  Authentications                        $2.08 
 ├─ Authentications (next 4000 Authentications)                                          4,000  Authentications                        $6.16 
 ├─ Authentications (over 5000 Authentications)                                         10,000  Authentications                        $7.70 
 └─ Advanced Authentications (first 10 Authentications)                                     10  Authentications                        $0.20 
 └─ Advanced Authentications (over 10 Authentications)                                  19,990  Authentications                      $199.90 
                                                                                                                                             
 ibm_resource_instance.resource_instance_kms                                                                                                 
 ├─ Key versions free allowance (first 5 Key Versions)                                       5  Key Versions                           $0.00 
 └─ Key versions                                                                           295  Key Versions                         $295.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_logdna_7day                                                                                         
 └─ Gigabyte Months                                                                         10  Gigabyte Months                       $15.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_logdna_7day_no_usage                                                                                
 └─ Gigabyte Months                                                      Monthly cost depends on usage: $1.50 per Gigabyte Months            
                                                                                                                                             
 ibm_resource_instance.resource_instance_logdna_lite                                                                                         
 └─ Lite plan                                                                                1                                         $0.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_monitoring_graduated                                                                                
 ├─ Additional Time series (first 1000 Time series hour)                                   100  Time series hour                       $0.01 
 ├─ Additional Time series (next 9000 Time series hour)                                      0  Time series hour                       $0.00 
 ├─ Additional Time series (next 90000 Time series hour)                                     0  Time series hour                       $0.00 
 ├─ Additional Time series (over 100000 Time series hour)                                    0  Time series hour                       $0.00 
 ├─ Additional Containers                                                                1,460  Container Hours                        $1.01 
 ├─ Additional API Calls                                                               100,000  API Calls                              $1.00 
 └─ Base Node Hour                                                                         730  Node Hours                            $35.04 
                                                                                                                                             
 ibm_resource_instance.resource_instance_monitoring_graduated_no_usage                                                                       
 ├─ Additional Time series (first 1000 Time series hour)                 Monthly cost depends on usage: $0.0001111111 per Time series hour   
 ├─ Additional Time series (next 9000 Time series hour)                  Monthly cost depends on usage: $0.0000694444 per Time series hour   
 ├─ Additional Time series (next 90000 Time series hour)                 Monthly cost depends on usage: $0.0000416667 per Time series hour   
 ├─ Additional Time series (over 100000 Time series hour)                Monthly cost depends on usage: $0.0000277778 per Time series hour   
 ├─ Additional Containers                                                Monthly cost depends on usage: $0.0006944444 per Container Hours    
 ├─ Additional API Calls                                                 Monthly cost depends on usage: $0.00001 per API Calls               
 └─ Base Node Hour                                                       Monthly cost depends on usage: $0.048 per Node Hours                
                                                                                                                                             
 ibm_resource_instance.resource_instance_monitoring_lite                                                                                     
 └─ Lite plan                                                                                1                                         $0.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_power_iaas                                                                                          
 └─ test                                                                                     1  Instance                               $0.00 
                                                                                                                                             
 ibm_resource_instance.resource_instance_secrets_manager                                                                                     
 ├─ Instance                                                                                 1  Instance                             $299.00 
 └─ Active Secrets                                                                         400  Secrets                               $80.00 
                                                                                                                                             
 OVERALL TOTAL                                                                                                                     $2,201.34 
──────────────────────────────────
17 cloud resources were detected:
∙ 17 were estimated, all of which include usage-based costs, see https://infracost.io/usage-file